projects/workforce-allocations-to-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/workforceallocations/AllocationDetail.kt

```diff
@@ -14,6 +14,7 @@ sealed interface AllocationDetail {
     val createdDate: ZonedDateTime
     val datasetCode: DatasetCode
     val code: String
+    val allocationReason: String?
```

Comment:
Can we make this be of type AllocationReason?  Then Jackson will automatically handle the mapping for you and you can remove the `fromTextOrDefaultInitial` method.

```suggestion
    val allocationReason: AllocationReason
```

Also, Gary confirmed they won't send `null` - they'll default to `INITIAL_ALLOCATION` for us: https://mojdt.slack.com/archives/C035YK9FFK4/p1765973148040839?thread_ts=1765963573.726079&cid=C035YK9FFK4